packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/service-discovery-integration.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpMethod, IVpcLink } from '@aws-cdk/aws-apigatewayv2';
+import * as servicediscovery from '@aws-cdk/aws-servicediscovery';
+import { HttpPrivateIntegration } from './http-private';
+
+/**
+ * Properties to initialize `HttpServiceDiscoveryIntegration`.
+ */
+export interface HttpServiceDiscoveryIntegrationProps {
+  /**
+   * The discovery service used for the integration
+   */
+  readonly service: servicediscovery.Service;
+
+  /**
+   * The vpc link to be used for the private integration
+   */
+  readonly vpcLink: IVpcLink;
```

Comment:
curious why `vpcLink` is mandatory here and cannot use the same logic as the other two?

Reply:
There is currently no way to extract out the underlying vpc from the `Service` construct. It should ideally come from
Service -> Namespace -> VPC
This is currently not present in the discovery module. We can update the current implementation when we have that.

Reply:
That's fine with me.